zanata-common-api/src/main/java/org/zanata/common/DocumentType.java

```diff
@@ -20,16 +20,20 @@
  */
 package org.zanata.common;
 
-import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Collections.unmodifiableSet;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 public enum DocumentType {
-    GETTEXT_PORTABLE_OBJECT("po"), GETTEXT_PORTABLE_OBJECT_TEMPLATE("pot"),
+
+    GETTEXT_PORTABLE_OBJECT(buildPotMap()), GETTEXT_PORTABLE_OBJECT_TEMPLATE(buildPotMap()),
```

Comment:
Does it still make sense to treat these as two separate document types? Probably better to make them a single type since it has both source and target extensions in it. Probably just `GETTEXT_PORTABLE_OBJECT`.